src/setup.js

```diff
@@ -24,17 +24,25 @@ for (var i = 0, l = pixelFunc.queue.length; i < l; i++) {
   pixelFunc.process.apply(pixelFunc, pixelFunc.queue[i]);
 }
 
-window.addEventListener('unload', function() {
-  if (!Config.pageCloseOnce) {
+// https://github.com/GoogleChromeLabs/page-lifecycle/blob/master/src/Lifecycle.mjs
+// Safari does not reliably fire the `pagehide` or `visibilitychange`
+var isSafari = typeof safari === 'object' && safari.pushNotification;
+
+// IE9-10 do not support the pagehide event, so we fall back to unload
+// pagehide event is more reliable but less broad than unload event for mobile and modern browsers
+var pageCloseEvent = 'onpageshow' in self && !is_safari ? 'pagehide' : 'unload';
+
+if (!Config.pageCloseOnce) {
```

Comment:
Is there a reason you switched the order of this from inside the event listener to outside? It's intention is to set up the `window.addEventListener` but then to make sure it only happens once we check inside the closure that it isn't trying to be called multiple times.

Reply:
You are correct, it's totally my fault. I am gonna revert it back. I just thought that it would not be a best practice to always bind an event listener even `pageCloseOnce` is true (maybe in a single page application).